File: server/src/Task/entity/value_object/Description.ts
export default class Description {
  private readonly value:string;
  constructor(value: string) {
    if (!value.trim()) {
      throw new Error("Description cannot be empty or blank");
    }
    if (value.length > 500) {
      throw new Error("Description is too long (max 500 characters)");
    }
    // if (!isMeaningful(value)) {
    //   throw new Error("Description is not clear enough");
    // }
    this.value = value;
  }

  public static fromString(desc: string): Description {
    return new Description(desc);
  }

  public toString(): string{
    return this.value;
  }
}

File: lib/model/value object/description.dart
class Description {
  final String value;

  Description(this.value) {
<<<<<<< HEAD
<<<<<<< HEAD
=======
    if (value.trim().isEmpty) {
      throw ArgumentError("Description cannot be empty or blank");
    }
>>>>>>> 19e03416083a52dbc55c65818d121799ce284671
=======
>>>>>>> master
    if (value.length > 500) {
      throw ArgumentError("Description is too long (max 500 characters)");
    }
  }

  factory Description.fromString(String desc) {
    return Description(desc);
  }
<<<<<<< HEAD

=======
  bool get isNotEmpty => value.isNotEmpty;
  bool get isEmpty => value.isEmpty;
  
>>>>>>> master
  @override
  String toString() {
    return value;
  }
}
